README.md

```diff
@@ -1,6 +1,197 @@
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+## Table of contents
+
+- [Example](#example)
+- [Use cases](#use-cases)
+- [Objectives](#objectives)
+- [Details](#details)
+  - [Privacy threat model and restrictions](#privacy-threat-model-and-restrictions)
+  - [Permissions and policies](#permissions-and-policies)
+  - [Rendering](#rendering)
+  - [Interactivity](#interactivity)
+  - [Accessbility](#accessbility)
+  - [Session history, navigation, and bfcache](#session-history-navigation-and-bfcache)
+- [Summary of differences between portals and iframes](#summary-of-differences-between-portals-and-iframes)
+- [Alternatives considered](#alternatives-considered)
+- [Acknowledgments](#acknowledgments)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Example
+
+A document can include a `<portal>` element, which renders the specified URL in a portal context:
+
+```html
+<portal id="myPortal" src="https://www.example.com/"></portal>
+```
+
+This is somewhat like an iframe, in that it provides a rendering of the specified document at `https://example.com/`. However, a portal is much more restricted than an iframe: user interaction with it is [limited](#interactivity) to just clicks (like a button or link), and capabilities like storage and cross-origin communication are [prohibited](#privacy-threat-model-and-restrictions) while rendering in the portal context. This allows portals to be used as a more-restricted view onto another document when the full capabilities of an iframe are not needed.
+
+In exchange for these restrictions, portals gain an additional capability that iframes do not have. A portal can be *activated*, which causes the embedding window to navigate, replacing its document with the portal:
+
+```js
+myPortal.activate();
+```
+
+TODO are we still thinking of making this the default click action? If so we should state that here. If it's still under debate we should state that it's under debate and link to an open issue.
+
+At this point, the user will observe that their browser has navigated to `https://www.example.com/` (e.g., via changes to the URL bar contents and back/forward UI). Since `https://example.com/` was already loaded in the portal context, this navigation will occur seamlessly and instantly, without a network round-trip or document re-initialization.
+
+For more advance use cases, the `https://www.example.com/` document can react to activation, using the `portalactivate` event. It can use this event to adapt itself to its new context, and even to adopt its *predecessor* (the document which previously occupied the tab) into a new portal context.
+
+```js
+window.addEventListener('portalactivate', e => {
+  document.body.classList.add('displayed-fully');
+  document.requestStorageAccess();
```

Comment:
I threw in some extra lines here because the code seemed uncompelling (and hopefully unnecessary?) if it was basically just "keep me around to work around the lack of bfcache". Suggestions for better stuff welcome.

Reply:
Done. More ideas still welcome.